lib/kubeclient/common.rb

```diff
@@ -51,6 +53,9 @@ def initialize_client(
       validate_auth_options(auth_options)
       handle_uri(uri, path)
 
+      @class_owner = class_owner
+      @entities = nil
+      @discovered = false
```

Comment:
@moolitayer isn't `@discovered` the same as `@entities.empty?`? (provided that you initialize `@entities` to an hash). If you really want to keep it explicit then you can have a method:

```ruby
    def discovered
      @entities.empty?
    end
```